build-tools/automation/yaml-templates/apk-instrumentation.yaml

```diff
@@ -1,13 +1,14 @@
 parameters:
-  configuration: []
-  testName: []
-  project: []
+  configuration: ""
+  testName: ""
+  project: ""
```

Comment:
These first three parameters were set to `[]` by default so you get an error if you forget to set them when using this tempalte. I think that is correct, @pjcollins?

Reply:
I think when I originally authored this the usage of `[]` was accidental. As far as I can tell, `[]` is intended to represent an empty array, yet these parameters are strings. I believe we need this change, but I'm not sure why the `[]` syntax didn't produce an error earlier. Azure Pipelines must cache our YAML in some way and the AZP syntax checking must have changed in the somewhat recent past.

There doesn't seem to be a clean way to specify that a template parameter is required. The documentation suggests using a [somewhat convoluted solution](https://docs.microsoft.com/en-us/azure/devops/pipelines/process/templates?view=azure-devops#required-parameters) for this.